src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -177,22 +198,26 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     ),
                 'required' => false,
             ])
-            ->add('cover_image', FileType::class, [
+            ->add('cover_image', CategoryCoverImageType::class, [
                 'label' => $this->trans('Category cover image', 'Admin.Catalog.Feature'),
                 'help' => $this->trans('This is the cover image for your category: it will be displayed on the category\'s page. The description will appear in its top-left corner.', 'Admin.Catalog.Help'),
                 'required' => false,
+                'preview_image' => $coverImage,
```

Comment:
Regardless of the proposition I made upper, maybe it would be simpler to pass the thumbnail URLs as the data so you wouldn't even need to rely on the `FormOptionsProviderInterface` This is what is done in other form types that use `ImagePreviewType`

But the main difference is that `ImagePreviewType` is used purely as a display item, whereas you also need to handle the upload of files so not sure the string data of a URL is really compatible with the FileType

One last possibility would be to use two form types:
- one for preview that in which the URL data is inserted for display
- one for file upload that is only used on submit